### PR TITLE
Migrate from aabb_quadtree to rstar

### PR DESCRIPTION
**Pull request by dabreegster**
`aabb_quadtree` is some ancient crate that's not even on Github! A/B Street has been using it since The Beginning. This PR finally migrates to the modern georust `rstar`.

I briefly tested map importing and the UIs, and didn't notice any major behavioral or performance changes. But it's possible something will come up the next time we regenerate all data or roll a release.

### CODE REVIEW

File: apps/map_editor/src/model.rs [hunk 1]
```diff
@@ -145,26 +145,6 @@ impl Model {
 
         self.recreate_world(ctx, &mut Timer::throwaway());
     }
-
-    fn compute_bounds(&self) -> Bounds {
```
Comment (dabreegster) [inline apps/map_editor/src/model.rs:149]: It's very nice to not need to figure out `Bounds` upfront for `World` and `FindClosest`

File: geom/src/bounds.rs [hunk 2]
```diff
@@ -92,6 +94,7 @@ impl Bounds {
     }
 
     /// Converts the boundary to the format used by `aabb_quadtree`.
+    // TODO On its way out
```
Comment (dabreegster) [inline geom/src/bounds.rs:97]: There's a circular dependency with osm2streets. I'll merge this PR and sort it out.

File: geom/src/bounds.rs [hunk 3]
```diff
@@ -105,6 +108,14 @@ impl Bounds {
         }
     }
 
+    /// Converts the boundary to the format used by `rstar`.
+    pub fn as_bbox2<T>(&self, data: T) -> GeomWithData<Rectangle<[f64; 2]>, T> {
```
Comment (dabreegster) [inline geom/src/bounds.rs:112]: ... And rename this

File: geom/src/bounds.rs [hunk 4]
```diff
@@ -253,3 +264,81 @@ impl GPSBounds {
...
+            .remove_with_selection_function(Selector(data))
+            .map(|item| item.data)
+    }
+}
+
+struct Selector<T>(T);
+
+impl<T: PartialEq> SelectionFunction<GeomWithData<Rectangle<[f64; 2]>, T>> for Selector<T> {
+    fn should_unpack_parent(&self, _envelope: &AABB<[f64; 2]>) -> bool {
+        // TODO Maybe we can ask the caller to estimate where the previous object was?
```
Comment (dabreegster) [inline geom/src/bounds.rs:318]: This is slower than the previous implementation -- `O(n)`. Not many uses of it.

File: importer/src/seattle.rs [hunk 5]
```diff
@@ -157,21 +155,20 @@ pub fn match_parcels_to_buildings(map: &mut Map, shapes: &ExtraShapes, timer: &m
...
     let mut units_per_bldg: Vec<(BuildingID, usize)> = Vec::new();
     timer.start_iter("match buildings to parcels", map.all_buildings().len());
     for b in map.all_buildings() {
         timer.next();
         // If multiple parcels contain a building's center, just pick one arbitrarily
-        for (idx, _, _) in quadtree.query(b.polygon.get_bounds().as_bbox()) {
-            let idx = *idx;
+        for idx in quadtree.query_bbox(b.polygon.get_bounds()) {
             if used_parcels.contains(&idx)
                 || !parcels_with_housing[idx].0.contains_pt(b.label_center)
```
Comment (dabreegster) [inline importer/src/seattle.rs:173]: As the TODO above suggests, it'd be potentially nice to store the entire `Polygon` and have a convenience method for finding the thing that contains some point.

**Comment by Robinlovelace** [comment]
Impressive!

**Review by michaelkirk — COMMENTED** [review]
Nice to see more use of the rstar crate - I'm glad it worked for abstreet!

**Comment by dabreegster** [comment]
> Nice to see more use of the rstar crate - I'm glad it worked for abstreet!

I still want to pursue the goal of making `geom` not exist at all, or be a very thin wrapper on `geo`, just haven't had enough time to focus on it.

I've been revisiting the purpose of the assertions in `Ring` that no duplicate points exist and in `PolyLine` that adjacent points aren't too close together. I might try some ideas and CC you "soon" if it's OK.

**Pull request merged**